xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-page/src/test/java/org/xwiki/test/page/PageTest.java

```diff
@@ -175,6 +180,31 @@ protected String renderPage(DocumentReference reference) throws Exception
         return doc.getRenderedContent(this.context);
     }
 
+    /**
+     * Load the provided document reference, render the loaded document and parse the result using {@link Jsoup}.
+     *
+     * @param reference the reference of the Document to load, render, and parse (and thus load from the
+     *     Classloader)
+     * @return the result of the parsing of the rendered result using {@link Jsoup}
+     * @throws Exception in case of errors
+     */
+    protected Document renderPageDocument(DocumentReference reference) throws Exception
+    {
+        return Jsoup.parse(renderPage(reference));
+    }
+
+    /**
+     * Render the provided document and parse the result using {@link Jsoup}.
+     *
+     * @param document the document to render and parse
+     * @return the result of the parsing of the rendered result using {@link Jsoup}
+     * @throws Exception in case of errors
+     */
+    protected Document renderPageDocument(XWikiDocument document) throws Exception
+    {
```

Comment:
why do we  need 2 signatures?

Reply:
IMO we have the two use cases as tests writers, either directly rendering a document by its reference, or first initializing a document in the test and only render and parse it.
Otherwise we can replace the two method by a `parseXML(Sting)` method.
WDYT?

Reply:
The goal of `renderPage`(and the goal of PageTest) is to load and render an XML page. If you don't do that then I'm not even sure why you need to extend `PageTest` vs using `@OldCoreTest`. Any idea?